app/controllers/datasets_controller.rb

```diff
@@ -1,2 +1,107 @@
 class DatasetsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @dataset = Dataset.new
+  end
+
+  def create
+    @dataset = Dataset.new(params.require(:dataset).permit(:id, :title, :summary, :description))
+    @dataset.organisation = current_user.primary_organisation
+
+    if @dataset.save
+      redirect_to new_licence_dataset_path(@dataset)
+    else
+      render 'new'
+    end
+  end
+
+  def licence
+    @dataset = current_dataset
+
+    if request.post?
+      licence = get_licence(params.require(:dataset).permit(:licence, :licence_other))
+      @dataset.licence = licence
+
+      redirect_to new_location_dataset_path(@dataset) if @dataset.save
+    end
+  end
+
+  def location
+    @dataset = current_dataset
+
+    if request.post?
+      location_params = params.require(:dataset).permit(:location1, :location2, :location3)
+      @dataset.update_attributes(location_params)
+
+      redirect_to new_frequency_dataset_path(@dataset) if @dataset.save
+    end
+  end
+
+  def frequency
+    @dataset = current_dataset
+
+    if request.post?
+      @dataset.frequency = params.require(:dataset).permit(:frequency)[:frequency]
+
+      redirect_to new_addfile_dataset_path(@dataset) if @dataset.save
+    end
+  end
+
+  def addfile
+    @dataset = current_dataset
+    @datafile = Datafile.new
+
+    if request.post?
+      file_params = params.require(:datafile).permit(:url, :name)
+      @datafile = Datafile.new(file_params)
+      @datafile.dataset = @dataset
+
+      if @datafile.save
```

Comment:
Understood -- I'll include this in a seperate issue with the changes to frequency